### PR TITLE
Expose parameter group interface

### PR DESCRIPTION
**Pull request by kenhkan**
I need a way to access parameter group as described in [torch.optim](https://docs.pytorch.org/docs/stable/optim.html) to implement per-parameter weight decay in AdamW. I couldn't find an existing way to do that that is in line with the PyTorch convention. This is a draft to showcase an implementation that works in my case. Is there a better way to do this with Hasktorch's conventions?

### CODE REVIEW

**Review by junjihashimoto — APPROVED** [review]

**Comment by junjihashimoto** [comment]
@kenhkan  Sorry for the late reply. I don't know of a better way, but I'd like you to try it out and I'll immediately incorporate any feedback you give me.
